calcom/cal.com
# fix: Make sure if header is set already, it isn't set again

## What does this PR do?

Fixes the following error. Though as far as I tested, the error arises only when there was already an error in the callback endpoint

<img width="536" alt="Screenshot 2023-11-02 at 10 26 12 AM" src="https://github.com/calcom/cal.com/assets/1780212/9cbf4e07-5c7d-4b01-aadf-b3a852ac0571">

In case of zohocalendar, the error was coming when this line was executed https://github.com/calcom/cal.com/blob/0091ecf572c063e899ec664e2e3f6aa6e1c6499c/packages/app-store/zohocalendar/api/callback.ts#L58. It returned a non nullish value causing the bug in `defaultResponder`.

## Type of change

## How should this be tested?
- Go to http://app.cal.local:3000/api/integrations/zohocalendar/callback?state=%7B%22returnTo%22%3A%22http%3A%2F%2Fapp.cal.local%3A3000%2Fapps%2Finstalled%22%7D&code=1000.86ee71b55ddb790a73d28b386032e781.1190f4c17b3b5fce09eae6121c8fc4b9&location=us and observe the header already set error. Shouldn't come with this branch

## Mandatory Tasks


## Checklist
- I haven't added tests that prove my fix is effective or that my feature works
